File: server/subscriptionsWS/data/subscriptionsDB/BLL/moviesBLL.py
from DAL.moviesDAL import MoviesDAL


class MoviesBLL:
    def __init__(self):
        self.__movies = MoviesDAL()

    # get all movies
    def get_all_movies(self):
        movies = list(self.__movies.get_all_movies())
        return movies

    # # get movie by id
    # def get_movie_by_id(self, id):
    #     movies = self.get_all_movies()
    #     movie_exists = list(filter(lambda m: str(m["_id"]) == id, movies))
    #     if not movie_exists:
    #         return "movie is not exists"
    #     movie = self.__movies.get_movie_by_id(id)
    #     return movie

    # # add movie
    # def add_movie(self, movie):
    #     movies = self.get_all_movies()
    #     movie_exists = list(filter(lambda m: str(m["_id"]) == id, movies))
    #     if movie_exists:
    #         return "movie is already exists"
    #     self.__movies.add_movie(movie)
    #     return movie

    # # update movie
    # def update_movie(self, id, movie):
    #     movies = self.get_all_movies()
    #     movie_exists = list(filter(lambda m: str(m["_id"]) == id, movies))
    #     if not movie_exists:
    #         return "movie is not exists"
    #     updated_movie = self.__movies.update_movie(id, movie)
    #     return updated_movie

    # # delete movie
    # def delete_movie(self, id):
    #     movies = self.get_all_movies()
    #     movie_exists = list(filter(lambda m: str(m["_id"]) == id, movies))
    #     if not movie_exists:
    #         return "movie is not exists"
    #     self.__movies.delete_movie(id)
    #     return f"The movie with ID {id} DELETED"

File: server/subscriptionsWS/BLL/moviesBLL.js
const MovieModel = require("../models/movieModel")

//get all movies
async function getAllMovies() {
    const movies = await MovieModel.find({})
    return movies
}

//get movie by id
async function getMovieById(id) {
    const movie = await MovieModel.findById(id)
    return movie
}

//add movie
async function addMovie(movie) {
    const newMovie = new MovieModel(movie)
    await newMovie.save()
    return newMovie
}

//update movie
async function updateMovie(id, movie) {
    await MovieModel.findByIdAndUpdate(id, movie)
    const updatedMovie = getMovieById(id)
    // console.log(updatedMovie)
    return updatedMovie
}

//delete movie
async function deleteMovie(id) {
    await MovieModel.findByIdAndDelete(id)
    return `the movie whis ID ${id} DELETED`
}

module.exports = { getAllMovies, getMovieById, addMovie, updateMovie, deleteMovie }


